TranslationAssistant.Business/DocumentTranslationManager.cs

```diff
@@ -54,10 +72,16 @@ public class DocumentTranslationManager
         /// <param name="isDir">The is dir.</param>
         /// <param name="sourceLanguage">The source language.</param>
         /// <param name="targetLanguage">The target langauge.</param>
-        public static void DoTranslation(string path, bool isDir, string sourceLanguage, string targetLanguage, bool ignoreHidden = false)
+        public static DocumentTranslatorResult DoTranslation(string path, bool isDir, string sourceLanguage, string targetLanguage, bool ignoreHidden = false)
         {
-            GetAllDocumentsToProcess(path, targetLanguage)
-                .ForEach(t => DoTranslationInternal(t, sourceLanguage, targetLanguage, ignoreHidden));
+            DocumentTranslatorResult result = new DocumentTranslatorResult();
+            foreach (string t in GetAllDocumentsToProcess(path, targetLanguage))
+            {
+                int[] resultArray = DoTranslationInternal(t, sourceLanguage, targetLanguage, ignoreHidden);
```

Comment:
I didn't realise this supports translating multiple files at once. On one hand I would say return `IEnumerable<DocumentTranslatorResult>`, but for our usage we will only ever have one 🤔 

`t` should be the full path to each file, so I think we could add a `FilePath` property to DocumentTranslatorResult. Then it should be fine to return IEnumerable.